src/snowflake/cli/plugins/spcs/image_repository/manager.py

```diff
@@ -51,8 +51,26 @@ def get_repository_api_url(self, repo_url):
 
         return f"{scheme}://{host}/v2{path}"
 
-    def create(self, name: str):
+    def create(
+        self,
+        name: str,
+        if_not_exists: bool,
+        replace: bool,
+    ):
+        if if_not_exists and replace:
+            raise ValueError(
+                "'replace' and 'if_not_exists' options are mutually exclusive for ImageRepositoryManager.create"
+            )
```

Comment:
Ideally we can figure out a way to resolve this earlier and just pass in some `CreateMode` enum (see also my previous comment). Maybe this isn't so possible in Typer...

Reply:
you're right that on the manager interface side it makes more sense to just use an enum.

Reply:
This chunking strategy makes sense to me.